### PR TITLE
Split the prepare properties to independent function for more readable.

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

**Review by yixinglu — DISMISSED** [review]

File: src/validator/FetchEdgesValidator.cpp [hunk 1]
```diff
@@ -139,102 +139,113 @@ Status FetchEdgesValidator::prepareEdges() {
...
+    props_.emplace_back(std::move(prop));
+    return Status::OK();
+}
+
+Status FetchEdgesValidator::preparePropertiesWithoutYield() {
+    // no yield
+    storage::cpp2::EdgeProp prop;
+    prop.set_type(edgeType_);
+    std::vector<std::string> propNames;   // filter the type
+    propNames.reserve(3 + schema_->getNumFields());
```
Comment (laura-ding) [inline src/validator/FetchEdgesValidator.cpp:223]: define `auto size = 3 + schema_->getNumFields()`

File: src/validator/FetchEdgesValidator.cpp [hunk 2]
```diff
@@ -139,102 +139,113 @@ Status FetchEdgesValidator::prepareEdges() {
...
+    outputs_.reserve(3 + schema_->getNumFields());
+    colNames_.reserve(3 + schema_->getNumFields());
+    geColNames_.reserve(3 + schema_->getNumFields());
+    // insert the reserved properties be compatible with 1.0
+    propNames.emplace_back(kSrc);
+    outputs_.emplace_back(edgeTypeName_ + "." + kSrc, Value::Type::STRING);
+    colNames_.emplace_back(edgeTypeName_ + "." + kSrc);
+    geColNames_.emplace_back(colNames_.back());
```
Comment (laura-ding) [inline src/validator/FetchEdgesValidator.cpp:231]: colNames_.emplace_back(edgeTypeName_ + "." + kSrc);
    outputs_.emplace_back(colNames_.back(), Value::Type::STRING);
    geColNames_.emplace_back(colNames_.back());

File: src/validator/FetchVerticesValidator.cpp [hunk 3]
```diff
@@ -106,126 +108,133 @@ Status FetchVerticesValidator::prepareVertices() {
...
-                        SchemaUtil::propTypeToValueType(tagSchema.second->getFieldType(i)));
-                    colNames_.emplace_back(tagName + "." + tagSchema.second->getFieldName(i));
-                    gvColNames_.emplace_back(colNames_.back());
-                }
+            // Check is prop name in schema
+            if (schema_->getFieldIndex(*expr->prop()) < 0) {
+                LOG(ERROR) << "Unknown column `" << *expr->prop() << "' in tag `"
+                            << tagName_ << "'.";
+                return Status::Error("Unknown column `%s' in tag `%s'.",
+                                        expr->prop()->c_str(),
```
Comment (laura-ding) [inline src/validator/FetchVerticesValidator.cpp:162]: alignment

File: src/validator/FetchVerticesValidator.cpp [hunk 4]
```diff
@@ -106,126 +108,133 @@ Status FetchVerticesValidator::prepareVertices() {
...
-                allTagsSchema;
-            allTagsSchema.reserve(allTags.size());
-            for (const auto &tag : allTags) {
-                allTagsSchema.emplace_back(tag.first, tag.second.back());
+            ExpressionUtils::rewriteLabelAttribute<TagPropertyExpression>(col->expr());
+        }
+        const auto *invalidExpr = findInvalidYieldExpression(col->expr());
+        if (invalidExpr != nullptr) {
+            return Status::Error("Invalid yield expression `%s'.",
+                                    col->expr()->toString().c_str());
```
Comment (laura-ding) [inline src/validator/FetchVerticesValidator.cpp:147]: alignment

File: src/validator/FetchVerticesValidator.cpp [hunk 5]
```diff
@@ -106,126 +108,133 @@ Status FetchVerticesValidator::prepareVertices() {
...
-            new YieldColumn(new InputPropertyExpression(new std::string(VertexID)),
-                            new std::string(VertexID)));
-        for (auto col : yield->columns()) {
-            newYieldColumns_->addColumn(col->clone().release());
+        outputs_.emplace_back(VertexID, Value::Type::STRING);
+        colNames_.emplace_back(VertexID);
+        gvColNames_.emplace_back(VertexID);  // keep compatible with 1.0
+        for (std::size_t i = 0; i < schema_->getNumFields(); ++i) {
+            outputs_.emplace_back(schema_->getFieldName(i),
+                                    SchemaUtil::propTypeToValueType(schema_->getFieldType(i)));
```
Comment (laura-ding) [inline src/validator/FetchVerticesValidator.cpp:204]: alignment

File: src/validator/FetchVerticesValidator.cpp [hunk 6]
```diff
@@ -106,126 +108,133 @@ Status FetchVerticesValidator::prepareVertices() {
...
+        const auto allTagsResult = qctx_->schemaMng()->getAllVerTagSchema(spaceId_);
+        NG_RETURN_IF_ERROR(allTagsResult);
+        const auto allTags = std::move(allTagsResult).value();
+        std::vector<std::pair<TagID, std::shared_ptr<const meta::NebulaSchemaProvider>>>
+            allTagsSchema;
+        allTagsSchema.reserve(allTags.size());
+        for (const auto &tag : allTags) {
+            allTagsSchema.emplace_back(tag.first, tag.second.back());
+        }
+        std::sort(allTagsSchema.begin(), allTagsSchema.end(), [](const auto &a, const auto &b) {
```
Comment (laura-ding) [inline src/validator/FetchVerticesValidator.cpp:219]: Maybe you want to output order the same as creating schemas. but I don't think it's better to sort every time. you can make the cache is ordered.

File: src/validator/FetchEdgesValidator.cpp [hunk 7]
```diff
@@ -139,102 +139,113 @@ Status FetchEdgesValidator::prepareEdges() {
...
+    propNames.emplace_back(kRank);
+    outputs_.emplace_back(edgeTypeName_ + "." + kRank, Value::Type::INT);
+    colNames_.emplace_back(edgeTypeName_ + "." + kRank);
+    geColNames_.emplace_back(colNames_.back());
 
+    for (std::size_t i = 0; i < schema_->getNumFields(); ++i) {
+        propNames.emplace_back(schema_->getFieldName(i));
+        outputs_.emplace_back(schema_->getFieldName(i),
+                                SchemaUtil::propTypeToValueType(schema_->getFieldType(i)));
```
Comment (laura-ding) [inline src/validator/FetchEdgesValidator.cpp:244]: alignment

File: src/validator/FetchEdgesValidator.cpp [hunk 8]
```diff
@@ -139,102 +139,113 @@ Status FetchEdgesValidator::prepareEdges() {
...
-                }
-                propsName.emplace_back(*expr->prop());
-                geColNames_.emplace_back(*expr->sym() + "." + *expr->prop());
+            // Check is prop name in schema
+            if (schema_->getFieldIndex(*expr->prop()) < 0 &&
+                reservedProperties.find(*expr->prop()) == reservedProperties.end()) {
+                LOG(ERROR) << "Unknown column `" << *expr->prop() << "' in edge `"
+                            << edgeTypeName_ << "'.";
+                return Status::Error("Unknown column `%s' in edge `%s'",
+                                        expr->prop()->c_str(),
```
Comment (laura-ding) [inline src/validator/FetchEdgesValidator.cpp:201]: alignment

File: src/validator/FetchVerticesValidator.cpp (hunk 3 above)
Comment (Shylock-Hg) [reply src/validator/FetchVerticesValidator.cpp]: Fixed.

File: src/validator/FetchEdgesValidator.cpp (hunk 1 above)
Comment (Shylock-Hg) [reply src/validator/FetchEdgesValidator.cpp:222]: Not speed more.

File: src/validator/FetchEdgesValidator.cpp (hunk 2 above)
Comment (Shylock-Hg) [reply src/validator/FetchEdgesValidator.cpp]: Fixed.

File: src/validator/FetchVerticesValidator.cpp (hunk 6 above)
Comment (Shylock-Hg) [reply src/validator/FetchVerticesValidator.cpp:215]: The schema will add/drop.

**Review by laura-ding — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**